pkg/runctl/secret_manager.go

```diff
@@ -0,0 +1,102 @@
+package runctl
+
+import (
+	"github.com/SAP/stewardci-core/pkg/apis/steward/v1alpha1"
+	"github.com/SAP/stewardci-core/pkg/k8s"
+	secrets "github.com/SAP/stewardci-core/pkg/k8s/secrets"
+	runifc "github.com/SAP/stewardci-core/pkg/runctl/run"
+	"github.com/pkg/errors"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	klog "k8s.io/klog/v2"
+)
+
+type secretManager struct {
+	secretHelper secrets.SecretHelper
+}
+
+// NewSecretManager creates secrets in the run namesapce
+func NewSecretManager(secretHelper secrets.SecretHelper) runifc.SecretManager {
+	return &secretManager{
+		secretHelper: secretHelper,
+	}
+}
+
+// CopyAll copies the required secrets based on a pipelineRun to the run namespace
+func (s *secretManager) CopyAll(pipelineRun k8s.PipelineRun) (string, []string, error) {
+	imagePullSecretNames, err := s.copyImagePullSecretsToRunNamespace(pipelineRun)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "failed to copy image pull secrets")
+	}
+
+	pipelineCloneSecretName, err := s.copyPipelineCloneSecretToRunNamespace(pipelineRun)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "failed to copy pipeline clone secret")
+	}
+
+	_, err = s.copyPipelineSecretsToRunNamespace(pipelineRun)
+	if err != nil {
+		return "", nil, errors.Wrap(err, "failed to copy pipeline secrets")
+	}
+
+	return pipelineCloneSecretName, imagePullSecretNames, nil
+}
+
+func (s *secretManager) copyImagePullSecretsToRunNamespace(pipelineRun k8s.PipelineRun) ([]string, error) {
+	secretNames := pipelineRun.GetSpec().ImagePullSecrets
+	transformers := []secrets.SecretTransformer{
+		secrets.StripAnnotationsTransformer("tekton.dev/"),
+		secrets.StripAnnotationsTransformer("jenkins.io/"),
+		secrets.StripLabelsTransformer("jenkins.io/"),
+		secrets.UniqueNameTransformer(),
+	}
+	return s.copySecrets(pipelineRun, secretNames, secrets.DockerOnly, transformers...)
+}
+
+func (s *secretManager) copyPipelineCloneSecretToRunNamespace(pipelineRun k8s.PipelineRun) (string, error) {
+	secretName := pipelineRun.GetSpec().JenkinsFile.RepoAuthSecret
+	if secretName == "" {
+		return "", nil
+	}
+	repoServerURL, err := pipelineRun.GetPipelineRepoServerURL()
+	if err != nil {
+		// TODO: this method should not modify the pipeline run -> must be handled elsewhere
+		pipelineRun.UpdateMessage(err.Error())
+		pipelineRun.UpdateResult(v1alpha1.ResultErrorContent)
+		return "", err
```

Comment:
Ouch. This should be fixed! Actually only the reconciler should modify the PipelineRun resource object.

We may introduce a new error annotation like `pkg/errors/recoverability.go` that indicates the Steward error classification (`error_config`, `error_content`, `error_infra`).

Reply:
See https://github.com/SAP/stewardci-core/pull/165/commits/38ac2f590d6f7fc13b6c3e4adec595e3b8660b85